cmake/cpu_arch_flags.cmake

```diff
@@ -63,8 +63,20 @@ function(get_arch_flags architecture compiler)
 
     elseif ("${HOST_ARCH}" STREQUAL "a64fx")
         set(HAS_REDZONE OFF PARENT_SCOPE)
-	set(CPU_ARCH_FLAGS "-mcpu=a64fx" PARENT_SCOPE)
-     endif()
+        set(CPU_ARCH_FLAGS "-mcpu=a64fx" PARENT_SCOPE)
+
+    elseif ("${HOST_ARCH}" STREQUAL "apple-m1")
+        set(HAS_REDZONE OFF PARENT_SCOPE)
+        if (compiler MATCHES "GNU")
```

Comment:
What about non-GNU?

Reply:
Haven't checked yet ;)
The only real alternative is clang but for that OpenMP becomes rather painful because Apple doesn't include the runtime lib per default...

Reply:
According to godbolt, they may also have the `-mcpu=apple-m1` or `-mcpu=apple-m2` flags the last time I checked (cf. also #962)

Reply:
Clang has, but I have no way of testing it without changing how OpenMP is linked - definitely not worth the effort for now
GCC doesn't have these flags.